processor/src/handler.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 Bitwise IO
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use sawtooth_sdk::messages::processor::TpProcessRequest;
+use sawtooth_sdk::processor::handler::ApplyError;
+use sawtooth_sdk::processor::handler::TransactionContext;
+use sawtooth_sdk::processor::handler::TransactionHandler;
+
+use bond_common::addressing::*;
+
+const BOND_NAMESPACE: &str = "bond namespace";
+
+pub struct BondTransactionHandler {
+    family_name: String,
+    family_versions: Vec<String>,
+    namespaces: Vec<String>,
+}
+
+impl BondTransactionHandler {
+    pub fn new() -> BondTransactionHandler {
+        BondTransactionHandler {
+            family_name: "bond".to_string(),
+            family_versions: vec!["0.1".to_string()],
+            namespaces: vec![hash(&BOND_NAMESPACE, 6).to_string()],
+        }
+    }
+}
+
+impl TransactionHandler for BondTransactionHandler {
+    fn family_name(&self) -> String {
+        self.family_name.clone()
+    }
+
+    fn family_versions(&self) -> Vec<String> {
+        self.family_versions.clone()
+    }
+
+    fn namespaces(&self) -> Vec<String> {
+        self.namespaces.clone()
+    }
+
+    fn apply(
+        &self,
+        request: &TpProcessRequest,
+        context: &mut TransactionContext,
```

Comment:
Got a couple compiler warnings here: 
```
warning: unused variable: `request`
  --> src/handler.rs:55:9
   |
55 |         request: &TpProcessRequest,
   |         ^^^^^^^ help: consider using `_request` instead
   |
   = note: #[warn(unused_variables)] on by default

warning: unused variable: `context`
  --> src/handler.rs:56:9
   |
56 |         context: &mut TransactionContext,
   |         ^^^^^^^ help: consider using `_context` instead
```
I'd recommend using the suggested underscores for now, and then change them back when you are actually using the variables. That way we don't get warnings during the build process and the code is still correct.